src/main/java/Duke.java

```diff
@@ -1,10 +1,265 @@
+                                return new Response(Message.TASK_FOUND, tmp);
+                            } else {
+                                return new Response(Message.NO_TASK_ON_DATE, null);
+                            }
+                        } else {
+                            return new Response(Message.LIST, taskList);
+                        }
+                    } else {
+                        return new Response(Message.NO_TASK, null);
+                    }
+
+                case "todo":
+                    checkSufficientTokens(descriptionTokens, "todo");
+
+                    Task task = new Todo(reconstructDescription(descriptionTokens));
+                    taskList.add(task);
+                    return new Response(Message.ADD_TASK, task);
+
+                case "deadline":
+                    //Fallthrough
+                case "event":
+                    //Check if description is present. Throws exception if description missing.
+                    checkSufficientTokens(descriptionTokens, "deadline");
+                    //Date portion is missing as size commandToken array is 1.
+                    throw new InsufficientArgumentAelitaException("date");
+
+                default:
+                    //No command matches input.
+                    throw new InvalidCommandAelitaException();
+                }
+
+            } else {
+                //Check if there are descriptions. Throws exception if there isn't.
+                checkSufficientTokens(descriptionTokens, "deadline");
+
+                //Concat the description tokens back to one string
+                String description = reconstructDescription(descriptionTokens);
+
+                switch (descriptionTokens[0].toLowerCase()) {
+                case "deadline":
+                    LocalDate deadlineDate = LocalDate.parse(commandTokens[1].substring(3));
+                    Task deadline = new Deadline(description, deadlineDate);
+                    taskList.add(deadline);
+                    return new Response(Message.ADD_TASK, deadline);
+
+                case "event":
+                    String[] dateTime = extractDateTime(commandTokens[1]);
+                    LocalDate eventDate = LocalDate.parse(dateTime[0]);
+
+                    Task event = new Event(description, eventDate, dateTime[1], dateTime[2]);
+                    taskList.add(event);
+                    return new Response(Message.ADD_TASK, event);
+
+                default:
+                    //No command matches input.
+                    throw new InvalidCommandAelitaException();
+                }
+
+            }
+        } catch (EmptyInputAelitaException e) {
+            return new Response(Message.EMPTY_COMMAND, null);
+
+        } catch (IoAelitaException e) {
+            return new Response(Message.IO_ERROR, null);
+
+        } catch (InsufficientArgumentAelitaException e) {
+            switch (e.getMessage()) {
```

Comment:
Good job on handling all the error messages in a single place. But I would recommend extracting the code segment as a method that can generate a Response given e for more readability of the executeCommand method.